disnake/ext/commands/params.py

```diff
@@ -403,97 +436,205 @@ def to_option(self) -> Option:
+
+    if self_param:
+        kwargs[self_param] = cog
+    if inter_param:
+        kwargs[inter_param] = interaction
+
+    return kwargs
+
+
+async def run_injections(
+    injections: Dict[str, Injection], interaction: Interaction, *args: Any, **kwargs: Any
+) -> Dict[str, Any]:
+    """Run and resolve a list of injections"""
+
+    async def _helper(name: str, injection: Injection) -> Tuple[str, Any]:
+        return name, await call_param_func(injection.function, interaction, *args, **kwargs)
+
+    resolved = await asyncio.gather(*(_helper(name, i) for name, i in injections.items()))
+    return dict(resolved)
+
+
+async def call_param_func(
+    function: Callable, interaction: Interaction, *args: Any, **kwargs: Any
+) -> Any:
+    """Call a function utilizing ParamInfo"""
+    self_param, inter_param, paraminfos, injections = collect_params(function)
+    formatted_kwargs = format_kwargs(interaction, self_param, inter_param, *args, **kwargs)
+    formatted_kwargs.update(await run_injections(injections, interaction, *args, **kwargs))
+    kwargs = formatted_kwargs
+
+    for param in paraminfos:
         if param.param_name in kwargs:
-            kwargs[param.param_name] = await param.convert_argument(inter, kwargs[param.param_name])
+            kwargs[param.param_name] = await param.convert_argument(
+                interaction, kwargs[param.param_name]
+            )
         elif param.default is not ...:
-            kwargs[param.param_name] = await param.get_default(inter)
+            kwargs[param.param_name] = await param.get_default(interaction)
 
-    return kwargs
+    return await maybe_coroutine(safe_call, function, **kwargs)
+
+
+def expand_params(command: AnySlashCommand) -> List[Option]:
+    """Update an option with its params *in-place*
+
+    Returns the created options
+    """
+    params = collect_nested_params(command.callback)
+
+    # update connectors and autocompleters
+    for param in params:
+        if param.name != param.param_name:
+            command.connectors[param.name] = param.param_name
+        if param.autocomplete:
+            command.autocompleters[param.name] = param.autocomplete
+
+    # TODO: Apply stuff like GuildCommandInteraction
+
+    return [param.to_option() for param in params]
 
 
-# NOTE: This is not worth overloading anymore unless we take
-# an sqlmodel approach and create overloads dynamically using templating
 def Param(
     default: Any = ...,
     *,
     name: str = "",
-    desc: str = None,
     description: str = None,
     choices: Choices = None,
-    conv: Callable[[Interaction, Any], Any] = None,
     converter: Callable[[Interaction, Any], Any] = None,
-    autocomp: Callable[[Interaction, str], Any] = None,
     autocomplete: Callable[[Interaction, str], Any] = None,
     channel_types: List[ChannelType] = None,
     lt: float = None,
     le: float = None,
     gt: float = None,
     ge: float = None,
-    min_value: float = None,
-    max_value: float = None,
```

Comment:
I thought that you wanted to move all shortened kwargs to `**kwargs`, but `min_value` and `max_value` are certainly not the shortened names, are you sure that you want to move them?

Reply:
That is because it's weirder to see `lt` and `gt` with `min_value` and `max_value` rather than just `lt`, `gt`, `le`, `ge` all together.

We could also move the `xt -> xe` parsing into the `Param` function and make all of those shorter names be the aliases.

Reply:
After trying out a few alternatives I've come to the conclusion that the abstraction of `min_value` and `max_value` behind `le` and `ge` just feels nicer to use. The combination of `min_value` and `gt` in the signature looks just straight-up wack yo!